src/engine/enginebuffer.cpp

```diff
@@ -1242,6 +1246,25 @@ void EngineBuffer::postProcess(const int iBufferSize) {
 }
 
 void EngineBuffer::updateIndicators(double speed, int iBufferSize) {
+    // Explicitly invalidate the visual playposition so waveformwidgetrenderer
+    // clears the visual when passthrough was activated while a track was loaded.
+    // TODO(ronso0) Check if postProcess() needs to be called at all when passthrough
+    // is active -- if no, remove this hack.
+    if (m_passthroughEnabled) {
+        m_visualPlayPos->setInvalid();
+        if (!m_passthroughWasEnabled) {
+            // Disable CUE and Play indicators
+            m_pCueControl->resetIndicators();
```

Comment:
Can we move this block to slotPassthroughChanged() and get rid of m_passthroughWasEnabled?

Reply:
:man_facepalming: 
Yes :+1: